src/compiler/crystal/semantic/call_error.cr

```diff
@@ -438,7 +438,7 @@ class Crystal::Call
   end
 
   def def_full_name(owner, a_def, arg_types = nil)
-    Call.def_full_name(owner, a_def, arg_types = nil)
+    Call.def_full_name(owner, a_def, arg_types)
   end
```

Comment:
That was a bug it seems.

Reply:
Well then it should have a spec exposing it...

Reply:
The only uses of `Crystal::Call#def_full_name` are in `Crystal::Call#raise_abstract_method_must_be_implemented` and it doesn't supply `arg_types`.

In fact, none of the uses of the _class_ method `Crystal::Call.def_full_name` supply `arg_types` either, so this argument is always unused.

Reply:
@HertzDevil I see more, for instance `Call#append_matches` which uses `#append_def_full_name` under-the-hood. `Call#append_matches` is being called in several places with `arg_types` present.

Reply:
As in the other case, if this is a hidden bug, it should be handled in a separate PR, ideally with a spec.